[docs] Improve the Getting Started documentation content (#30808)
mui-org/material-ui: docs/src/pages/getting-started/faq/faq.md

  If you use MUI in a commercial project and would like to support its continued development by becoming a Sponsor, or in a side or hobby project and would like to become a Backer, you can do so through OpenCollective.
  All funds donated are managed transparently, and Sponsors receive recognition in the README and on the MUI home page.

> ## Why aren't my components rendering correctly in production builds?

Author: I've moved the things related to v4 on the end and added the `[v4]` tag

Reviewer: I think `[v4]` should be the prefix because it is easier to notice on global search.

Author: Fair enough, let me update

## I cannot use components as selectors in the styled() utility. What should I do?

If you are getting the error: `TypeError: Cannot convert a Symbol value to a string`, take a look at the [styled()](/system/styled/#how-to-use-components-selector-api) docs page for instructions on how you can fix this.

## Why aren't my components rendering correctly in production builds? [v4]

> The #1 reason this likely happens is due to class name conflicts once your code is in a production bundle.

Reviewer 2: ```suggestion
The #1 reason this happens is likely due to class name conflicts once your code is in a production bundle.
```

To correct this issue, all components on the page need to be initialized such that there is only ever **one class name generator** among them.

You could end up accidentally using two class name generators in a variety of scenarios:

- You accidentally **bundle** two versions of MUI. You might have a dependency not correctly setting MUI as a peer dependency.
- You are using `StylesProvider` for a **subset** of your React tree.
- You are using a bundler and it is splitting code in a way that causes multiple class name generator instances to be created.

> If you are using webpack with the [SplitChunksPlugin](https://webpack.js.org/plugins/split-chunks-plugin/), try configuring the [`runtimeChunk` setting under `optimizations`](https://webpack.js.org/configuration/optimization/#optimization-runtimechunk).

Overall, it's simple to recover from this problem by wrapping each MUI application with [`StylesProvider`](/styles/api/#stylesprovider) components at the top of their component trees **and using a single class name generator shared among them**.

> ### CSS works only on first load then is missing [v4]

Reviewer 2: ```suggestion
### CSS works only on first load and goes missing [v4]
```
